src/onnx/parse_dynamicquantizelinear.cpp

```diff
@@ -0,0 +1,151 @@
+Outputs Scale, ZeroPoint and Quantized Input for a given FP32 Input. Scale is calculated as:
+y_scale = (maximum(0, max(x)) - minimum(0, min(x))) / (qmax - qmin)
+* where qmax and qmin are max and min values for quantization range i.e. [0, 255] in case of uint8
+* data range is adjusted to include 0.
+
+Zero point is calculated as:
+intermediate_zero_point = qmin - min(x)/y_scale
+y_zero_point = cast(round(saturate(itermediate_zero_point)))
+* where qmax and qmin are max and min values for quantization range .i.e [0, 255] in case of uint8
+* for saturation, it saturates to [0, 255] if it's uint8, or [-127, 127] if it's int8. Right now
+only uint8 is supported.
+* rounding to nearest ties to even. Data quantization formula is:
+
+y = saturate (round (x / y_scale) + y_zero_point)
+* for saturation, it saturates to [0, 255] if it's uint8, or [-127, 127] if it's int8.Right now only
+uint8 is supported.
+* rounding to nearest ties to even.
+
+Version
+This version of the operator has been available since version 11 of the default ONNX operator set.
+
+Inputs
+x : T1
+Input tensor
+
+Outputs
+y : T2
+Quantized output tensor
+
+y_scale : tensor(float)
+Output scale. It's a scalar, which means a per-tensor/layer quantization.
+
+y_zero_point : T2
+Output zero point. It's a scalar, which means a per-tensor/layer quantization.
+
+Type Constraints
+T1 : tensor(float)
+Constrain 'x' to float tensor.
+
+T2 : tensor(uint8)
+Constrain 'y_zero_point' and 'y' to 8-bit unsigned integer tensor.
+*/
+
+struct parse_dynamicquantizelinear : op_parser<parse_dynamicquantizelinear>
+{
+    std::vector<op_desc> operators() const { return {{"DynamicQuantizeLinear"}}; }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd*/,
+                                       const onnx_parser& /*parser*/,
+                                       const onnx_parser::node_info& info,
+                                       const std::vector<instruction_ref>& args) const
+    {
+        auto x       = args[0];
+        auto x_shape = x->get_shape();
+        if(x_shape.dynamic())
+            MIGRAPHX_THROW("DYNAMICQUANTIZELINEAR: dynamic shapes are not supported");
+
+        auto x_type = x_shape.type();
+        // 1. Computing y_scale
+        auto l0 = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {0}});
+        // DynamicQuantizeLinear only has uint8 quantization
+        auto q_max = info.add_literal(
+            migraphx::literal{migraphx::shape{x_type}, {std::numeric_limits<uint8_t>::max()}});
+        auto q_min = info.add_literal(
+            migraphx::literal{migraphx::shape{x_type}, {std::numeric_limits<uint8_t>::min()}});
+        auto x_reshape = x;
+        if(not(x_shape.lens().size() == 1))
+        {
+            x_reshape = info.add_instruction(
+                migraphx::make_op("reshape", {{"dims", {x_shape.elements()}}}), x);
+        }
+        // maximum(0, max(x))
```

Comment:
Optional optimization. Is it possible to add a '0' to X, and bypass both the max() and min() comparisons..

Reply:
Thanks.